### PR TITLE
Pastas docs 1.0

### PR DESCRIPTION
**Pull request by raoulcollenteur**
# Short Description

This PR will update the Documentation for Pastas 1.0.

# Checklist before PR can be merged:
- [x] closes issue #180 and #408 
- [x] is documented
- [x] PEP8 compliant code
- [x] tests added / passed
- [x] Example Notebook (for new features)

### CODE REVIEW

**Comment by raoulcollenteur** [comment]
The new documentation (WIP) can be viewed here: https://pastas--502.org.readthedocs.build/en/502/index.html

**Comment by martinvonk** [comment]
I'm done with the examples part. I removed the gallery layout because it often fails and is complicated.

**Comment by raoulcollenteur** [comment]
The "About" section is ready from my part and ready ready for review.

**Comment by martinvonk** [comment]
> The "About" section is ready from my part and ready ready for review.

I like it :)

Maybe we can do three things:

1. Make the about section the last option in the header? I feel that is more natural. So first section User Guide, last About.
2. Add some whitespace above the eawag logo? Now it is feels a bit high:
image
3. Maybe remove the overlapping names in maintainers, developers and contributers?

File: doc/about/01_history.rst [hunk 1]
```diff
@@ -0,0 +1,37 @@
+History
+=======
+
+The Development of Pastas started in the spring of 2016 at the TU Delft (The Netherlands) and Artesia (The
+Netherlands). Researchers at the TU Delft required a flexible framework that could support future research on time
+series analysis of groundwater data. Consulting company `Artesia <https://www.artesia-water.nl>`_ on the other hand,
+needed a tool that allowed them to perform more complex analyses in a scripted environment. As such, Pastas was
+developed under a strong partnership between academia and private industry, a collaboration that continues until this
```
Comment (dbrakenhoff) [inline doc/about/01_history.rst:8]: "continues to this day"

File: doc/about/01_history.rst [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+development of Pastas. Since its inception, Pastas has been applied in over a dozen countries worldwide. Supporting
+international research continues to be an important goal of Pastas. For a list of peer-reviewed publications using
+Pastas please see the :doc:`Publications page <publications>`.
+
+Pastas 1.0
+----------
+
+In february 2023, Pastas version 1.0 was released, the celebration of 7 years of research and development of Pastas.
+Although Pastas was already operational for a couple of years, the application programming interface (API) was still
+changing sporadically and older (unused) model code was mounting. This made maintenance harder and harder. In Pastas
```
Comment (dbrakenhoff) [inline doc/about/01_history.rst:33]: harder and harder -> increasingly difficult?

Comment (martinvonk) [reply doc/about/01_history.rst:33]: harder better faster stronger?

Comment (martinvonk) [reply doc/about/01_history.rst]: see [4d26710](https://github.com/pastas/pastas/pull/502/commits/4d2671080dfc6660cf34aeb29c457440080e5530)

File: doc/about/01_history.rst (hunk 1 above)
Comment (martinvonk) [reply doc/about/01_history.rst]: [4d26710](https://github.com/pastas/pastas/pull/502/commits/4d2671080dfc6660cf34aeb29c457440080e5530)

**Comment by mbakker7** [comment]
In `01_getting_started` it still says "numba  #(large speed-up, highly recommended)". That 'highly recommended' can go, right? as it is now a requirement. So maybe also remove the 'large speed-up part'.

**Comment by martinvonk** [comment]
> In `01_getting_started` it still says "numba #(large speed-up, highly recommended)". That 'highly recommended' can go, right? as it is now a requirement. So maybe also remove the 'large speed-up part'.

I agree that `highly recommended` is a bit misleading since Numba is an installation requirement. However, users can turn off Numba by setting the global USE_NUMBA variable so in that sense, we do "recommend" to not do that :)

**Comment by mbakker7** [comment]
It is confusing to have options in the requirements. I suggest to leave it
out from the docs.

**Comment by raoulcollenteur** [comment]
The following sections are done from my part and can be reviewed and updated:

- [ ] About
- [ ] Developers
- [ ] Examples
- [ ] API Docs

To-do:

- [ ] Update code example on home page
- [ ] Write modeling workflow notebook (@raoulcollenteur)
- [ ] Write response functions notebook (@mbakker7 )
- [ ] finish stressmodels notebook (@martinvonk )
- [ ] Make benchmark section nice (@raoulcollenteur)

**Comment by dbrakenhoff** [comment]
I've taken a look at the sections you marked as done @raoulcollenteur. Looks good, I made a few minor changes.

**Review by dbrakenhoff — APPROVED** [review]

**Comment by martinvonk** [comment]
I copied some code from examples/02_fix_parameters.ipynb to fill the modeling workflow notebook.

**Review by martinvonk — APPROVED** [review]

**Pull request merged**